src/load_save.cpp

```diff
@@ -0,0 +1,84 @@
+#include <migraphx/load_save.hpp>
+#include <migraphx/json.hpp>
+#include <migraphx/msgpack.hpp>
+#include <fstream>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+std::vector<char> read_buffer(const std::string& filename)
+{
+    std::ifstream is(filename, std::ios::binary | std::ios::ate);
+    std::streamsize size = is.tellg();
+    is.seekg(0, std::ios::beg);
+
+    std::vector<char> buffer(size);
+    if(!is.read(buffer.data(), size))
+    {
+        MIGRAPHX_THROW("Error reading file: " + filename);
```

Comment:
I would suggest the error message as "Load program, error reading file" since we have "Error reading file:" message in other places.

Reply:
I added the function name to `MIGRAPHX_THROW`.